vexcl/reductor.hpp

```diff
@@ -326,28 +326,27 @@ Reductor<real,RDC>::operator()(const Expr &expr) const {
                             queue[d], source.str(), "vexcl_reductor_kernel"));
             } else {
                 source.smem_declaration<real>();
-                source.new_line() << type_name< shared_ptr<real> >() << " sdata = smem;";
```

Comment:
This does break the CUDA backend. In CUDA, as opposed to OpenCL, the shared memory is not passed as a kernel parameter, it is just declared as an array with `__shared__` qualifier. Compare implementation of `source.smem_declaration()` (used in the line above) for [OpenCL](https://github.com/ddemidov/vexcl/blob/7ac0b2516bc3317abf9893a5afb0828a05a93c6d/vexcl/backend/opencl/source.hpp#L197-L201) and for [CUDA](https://github.com/ddemidov/vexcl/blob/7ac0b2516bc3317abf9893a5afb0828a05a93c6d/vexcl/backend/cuda/source.hpp#L176-L180) backends.